evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -626,3 +625,80 @@ def test_simple_imputer_boolean_nullable_valid_train_empty_test():
     X_t = imp.transform(X_test)
     assert not X_t["a"].isna().any()
     assert isinstance(X_t.ww.logical_types["a"], BooleanNullable)
+
+
+def test_simple_imputer_all_bools_at_fit_and_transform():
+    """Confirms that the simple imputer can handle data with only the bool dtype
+    which sklearn would error on."""
+    X = pd.DataFrame(
+        {
+            "bools1": pd.Series([True, False, True, True] * 20),
+            "bools2": pd.Series([True, False, True, False] * 20),
+        },
+    )
+    X.ww.init(
+        logical_types={
+            "bools1": "Boolean",
+            "bools2": "Boolean",
+        },
+    )
+
+    imp = SimpleImputer(impute_strategy="most_frequent")
+    imp.fit(X)
+
+    X_imputed = imp.transform(X)
+    pd.testing.assert_frame_equal(X, X_imputed)
+
+
+def test_simple_imputer_all_bools_at_fit_and_transform_with_all_null_and_nl_cols(
+    imputer_test_data,
+):
+    """Confirm that the simple imputer, which doesn't pass all null or natural language columns
+    to sklearn works when the remaining columns are all teh bool dtype, which sklearn would error on.
+    """
+    X = imputer_test_data.ww[["all nan", "bool col", "natural language col"]]
+    X_copy = X.ww.copy()
+
+    imp = SimpleImputer(impute_strategy="most_frequent")
+    imp.fit(X)
+
+    X_imputed = imp.transform(X)
+    pd.testing.assert_frame_equal(X_copy.ww.drop("all nan"), X_imputed)
+
+
+def test_simple_imputer_all_bools_at_fit_with_nans_at_transform():
+    """Confirm that the simple imputer can handle data whose dtype is different at transform
+    when originally the data only had bool dtype columns."""
+    # X_train will be only bool dtypes so the _component_obj won't be fit
+    X_train = pd.DataFrame(
+        {
+            "bools1": pd.Series([True, False, True, True] * 20),
+            "bools2": pd.Series([True, False, True, False] * 20),
+        },
+    )
+    X_train.ww.init(
+        logical_types={
+            "bools1": "Boolean",
+            "bools2": "Boolean",
+        },
+    )
```

Comment:
Is it worth adding the all bools columns to the `imputer_test_data`

Reply:
Probably! 

I can add in a second all boolean no nulls column into the fixture that way I can get my "all bools" dataframe with a `select("Boolean")` call and then add nans in after fit to create the test data for this test. I can't think of anything that would be inherently problematic about adding a second `Boolean` column other than maybe some tests that only expect there to be one might fail.